src/Components/Browser.JS/src/Platform/Circuits/AutoReconnectCircuitHandler.ts

```diff
@@ -0,0 +1,48 @@
+import { CircuitHandler } from "./CircuitHandler";
+export class AutoReconnectCircuitHandler implements CircuitHandler {
+  modal: HTMLDivElement;
+  intervalHandle: number | null;
+
+  constructor(private maxRetries: number = 3, private retryInterval: number = 3000) {
+    const modal = document.createElement('div');
+    modal.className = "modal";
+    modal.appendChild(document.createTextNode("Attempting to reconnect to the server..."));
+    document.addEventListener("DOMContentLoaded", function (event) {
+      document.body.appendChild(modal);
+    });
+
+    this.modal = modal;
+    this.intervalHandle = null;
+  }
+  onConnectionUp() {
+    this.modal.style.display = 'none';
+    this.cleanupTimer();
+  }
+  async onConnectionDown() {
+    this.modal.style.display = 'block';
+    this.cleanupTimer();
+
+    let retries = 0;
+    this.intervalHandle = window.setInterval(async () => {
+      if (retries++ > this.maxRetries) {
+        this.cleanupTimer();
+      }
+
+      try {
+        await window['Blazor'].reconnect();
```

Comment:
I might change this to return a value rather than throw. This is kinda unwieldy

Reply:
Are we going to distinguish between "couldn't get an intelligible answer from the server, e.g., because we're offline or it returned a 500" and "server responded that the circuit ID is invalid"? In the first case it makes sense to keep retrying, whereas the second case is permanently fatal.

Again, I don't mind if this isn't attempted right now, but if not it would be good to collect these kinds of enhancement ideas for potential follow-up.

Reply:
Yup. I tweaked it a bit. If we get a definite cannot reconnect, we quit early now